### PR TITLE
initial project configuration

### PR DESCRIPTION
**Pull request by tplevko**
initial project configuration with vite, react, patternfly and storybook

added also two basic modals with stories for `Add Path` and `Add Tag`

### CODE REVIEW

File: package-lock.json
Comment (lordrip) [inline package-lock.json:1]: It looks like this file shouldn't be here :smile:

**Review by lordrip — APPROVED** [review]
Thanks a lot for putting this together @tplevko :bow: 

After checking, I think there are two open topics:

1. Upgrade `yarn` to the latest version (4.x)
2. Split the module in two sub modules:
    a. The components module, from where we're gonna export all components and also have a simple composition that we could see on GitHub pages

    b. The UI test / storybook module, from where we're gonna import the exported components from the previous module. The goal is to ensure that we're exporting the right components and that also the API is good.

I reckon that this changeset it's not trivial, si we could tackle them one by one in additional PRs

**Comment by EricWittmann** [comment]
Why use `yarn` instead of `npm`?  Any issues with productization doing that?

**Comment by EricWittmann** [comment]
Also can we apply Apicurio code styling if there are no objections (expecting there might be)?

* https://github.com/Apicurio/apicurio-registry/blob/main/ui/ui-app/package.json#L16
* https://github.com/Apicurio/apicurio-registry/blob/main/ui/ui-app/.eslintrc.cjs

Also we might want to rename `vite-config.ts` to `vite-config.mts` if we plan to use `vite-tsconfig-paths`.  I had to do this recently:

* https://github.com/Apicurio/apicurio-registry/pull/5007

**Comment by lordrip** [comment]
> Why use `yarn` instead of `npm`? Any issues with productization doing that?

@EricWittmann , nothing in terms of productization, mostly because handling multiple modules in a monorepo fashion is easier with `yarn` than `npm`. In the case of `npm`, we would need to use `lerna` or `nx` in order to orchestrate multiple modules and their dependencies.

yarn `1.x` vs `4.x` is mostly because `1.x` is deprecated.

That being said, I'm not against using `npm`, it's just a bit more work and that's it.

Please let us know what your preference is so we can "choose our destiny" :smile:

**Comment by lordrip** [comment]
> Also can we apply Apicurio code styling if there are no objections (expecting there might be)?

We could, we just need to pick one of the following scenarios:
1. Upgrade apicurio-registry eslint config to be compatible with `eslint 9.x`, since they changed the configuration file format
2. Downgrade `eslint` in this project to `8.x` in order to be compatible


> Also we might want to rename `vite-config.ts` to `vite-config.mts` if we plan to use `vite-tsconfig-paths`. 

Sure, we can. I'm not sure we're gonna use `vite-tsconfig-paths` at this point, but better to accommodate this scenario just in case

**Comment by EricWittmann** [comment]
> > Why use `yarn` instead of `npm`? Any issues with productization doing that?
> 
> @EricWittmann , nothing in terms of productization, mostly because handling multiple modules in a monorepo fashion is easier with `yarn` than `npm`. In the case of `npm`, we would need to use `lerna` or `nx` in order to orchestrate multiple modules and their dependencies.
> 
> yarn `1.x` vs `4.x` is mostly because `1.x` is deprecated.
> 
> That being said, I'm not against using `npm`, it's just a bit more work and that's it.
> 
> Please let us know what your preference is so we can "choose our destiny" 😄

Great thanks!  I have no preference, I was just curious.  Will this project utilize multiple modules?

(agree that multi-module with `npm` is terrible, I didn't know that `yarn` did it better)

**Comment by EricWittmann** [comment]
> > Also can we apply Apicurio code styling if there are no objections (expecting there might be)?
> 
> We could, we just need to pick one of the following scenarios:
> 
>     1. Upgrade apicurio-registry eslint config to be compatible with `eslint 9.x`, since they changed the configuration file format
> 
>     2. Downgrade `eslint` in this project to `8.x` in order to be compatible

Yeah let's upgrade apicurio's eslint config!  I've been holding back on upgrading to eslint 9.x in other projects because I haven't had time to do that.  If you know how, or can tell me how, then let's get that done!  Would be a nice improvement.

**Comment by lordrip** [comment]
@EricWittmann 

> Great thanks! I have no preference, I was just curious. Will this project utilize multiple modules?
> (agree that multi-module with npm is terrible, I didn't know that yarn did it better)

I think it would be convenient to have at least 2 modules, similar to what we have in Kaoto:
a. The main module exporting components
b. A testing module that imports the public components and adds stories and e2e.

There are a couple of advantages with the latter:
a. In case of security vulnerabilities, they will be properly reflected on the right package, for instance, if there's an issue with a test dependency, it will be reflected on this package, reducing the noise a bit.
b. We ensure that we export the right components and that also we keep the API stable, or said in a different way, if there's a breaking change in a component, this module will act as a close consumer, properly identifying when such unintended change happens.

But as I said, this is just a suggestion :smile:


> Yeah let's upgrade apicurio's eslint config! I've been holding back on upgrading to eslint 9.x in other projects because I haven't had time to do that. If you know how, or can tell me how, then let's get that done! Would be a nice improvement.

Hehe, me too, I haven't dedicated the time to update Kaoto either :sweat_smile:, I hope that I can take care this week

**Review by lordrip — APPROVED** [review]
LGTM

Pending items to tackle next:
1. Update apicurio code style so it can work with eslint 9.x
2. Use apicurio code style in this project
3. Confirm whether we're gonna follow the monorepo approach
4. Start with additional components

@EricWittmann, are we missing something?

**Comment by EricWittmann** [comment]
LGTM

**Pull request merged**